src/lib.rs

```diff
@@ -239,3 +241,59 @@ pub trait ReadFrom {
     where
         Self: Sized;
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_test::{assert_de_tokens, assert_de_tokens_error, Token};
+
+    #[test]
+    fn network_can_deserialize() {
+        assert_de_tokens(
+            &Network::MainNet,
+            &[Token::UnitVariant {
+                name: "Network",
+                variant: "mainnet",
+            }],
+        );
+        assert_de_tokens(
+            &Network::TestNet,
+            &[Token::UnitVariant {
+                name: "Network",
+                variant: "testnet",
+            }],
+        );
+        assert_de_tokens_error::<Network>(
+            &[Token::UnitVariant {
+                name: "Network",
+                variant: "other",
+            }],
+            "unknown variant `other`, expected `mainnet` or `testnet`",
+        );
+    }
+
+    #[test]
+    fn keytype_can_deserialize() {
+        assert_de_tokens(
+            &KeyType::Ed25519,
+            &[Token::UnitVariant {
+                name: "KeyType",
+                variant: "ed25519",
+            }],
+        );
+        assert_de_tokens(
+            &KeyType::EccCompact,
+            &[Token::UnitVariant {
+                name: "KeyType",
+                variant: "ecccompact",
```

Comment:
this ain't right the string representation for EccCompact is `ecc_compact`.. as encoded by `FromStr`.. You'l need a rename for that specific variant

Reply:
Fixed fd66457